### ISSUE TITLE
Cache decryption password while loopback file is in unlocked state

### ISSUE THREAD

**Issue by mystiquewolf** [systemd/systemd#21552]
**Is your feature request related to a problem? Please describe.**
The problem is that i have selected 4GB of memory for LUKS PBKDF function and i only have that much free before login. After login i have apps opened just like every regular user. So when a polkit asks for password the OOM killer throws a killing spree party.

**Describe the solution you'd like**
Just cache the password while loopback file is unlocked and don't try to decrypt just to check. 

**Describe alternatives you've considered**
The only other alternative i've considered is lowering the RAM required, but i much prefer not to. 

**The systemd version you checked that didn't have the feature you are asking for**
249.7-1

One other question i would like to ask is whether homectl update will reencrypt the luks data if i pass different luks options to it - that's because i want to know whether this profile like it is has any chance of being usable before i spend much more time customizing/initializing it.

EDIT: This also happens when you have a dozen applications opened in tty2/graphical and you switch to tty3 and try to login to the user (or potentially to a different systemd-homed user). IMO this is a big usability issue.

Let's say that we can't workaround if a different user is unlocked, but why does systemd need to unlock and already unlocked loop file?

**Comment by poettering** [comment]
So this is about homed? It's a feature, not a bug that we remove all passwords from memory when not needing them...

**Comment by mystiquewolf** [comment]
I guess it's a security tradeoff then. I don't really know if it's possible to evade this high memory usage on password checking with the mentioned LUKS parameters... If there's nothing to be done you can close this issue.

**Comment by poettering** [comment]
> I guess it's a security tradeoff then. I don't really know if it's possible to evade this high memory usage on password checking with the mentioned LUKS parameters... If there's nothing to be done you can close this issue.

Well, if there was that would defeat the point of the setting, no?

**Comment by mystiquewolf** [comment]
I'm not exactly competent about this, but doesn't this feature high-memory usage have its highest value before you've unlocked the loop file? After that i don't get it why you would need the high-memory usage. Probably because you don't want to keep the secret in memory.

**Comment by poettering** [comment]
Yes, homed is designed to be frugal with secret data: we don't want to keep sensitive stuff in memory we really don't need to keep around... And the assumption is that if we can reauth with original stored data we don't need unencrypted pws around.
